### PR TITLE
remove redundant Copy traits of EvalConfig and Tz, feature requires: const_int_ops;

### PR DESCRIPTION
**Pull request by iosmanthus**
* remove redundant `Copy` traits of `EvalConfig` and `Tz`
* remove feature `const_int_ops`, 

fix issue https://github.com/tikv/tikv/issues/4341 and https://github.com/tikv/tikv/issues/4342

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- Summarize your change (mandatory)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Separately describe each logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## What are the type of changes? (mandatory)
- Improvement (a change which is an improvement to an existing feature)
`EvalConfig` and `Tz` use Copy sematic be the default. But it consumes more resources than passing references of `Tz`. But lots of functions just accept `Tz` not the references. So, I remove the Copy trait and use move semantics. Clone the `Tz`  by need.

## How has this PR been tested? (mandatory)
I pass through the compiling with `rustc nightly 2019-03-10`.

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_string.rs [hunk 1]
```diff
@@ -903,28 +903,6 @@ impl ScalarFunc {
             Ok(Some(Cow::Borrowed(b"")))
         }
     }
-
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_string.rs]: why remove it?

File: src/lib.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@
 #![feature(integer_atomics)]
 #![feature(duration_as_u128)]
 #![feature(proc_macro_hygiene)]
-#![feature(const_int_ops)]
+//#![feature(const_int_ops)]
```
Comment (AndreMouche) [inline src/lib.rs]: Please remove it?

Comment (iosmanthus) [reply src/lib.rs]: sorry, I will fix and open another pr. I will close this one first.thx!

**Comment by iosmanthus** [comment]
> Could we split it into two PR? One PR -- one issue?

sorry, I will fix it and open another two prs. Please close this one.

**Comment by breeswish** [comment]
@iosmanthus You got it 😄There is pretty a lot of signatures to change, which is not a simple task, so I opened an issue first.... XD

**Comment by breeswish** [comment]
Close for https://github.com/tikv/tikv/pull/4355

**Pull request closed without merging**